include/internal/catch_generators.hpp

```diff
@@ -349,6 +349,53 @@ namespace Generators {
         );
     }
 
+    template <typename T>
+    class RangeGenerator final : public IGenerator<T> {
```

Comment:
I think that it should be limited only for arithmetic types.

Reply:
I personally prefer to keep it open for anything that provides the correct interface. This way someone can still use it with their own classes, such as a `BigInteger`. A rare use case, but I think trying to prevent someone from putting a string in it isn't going to help anyone either.

But that's my opinion, if you really want me to I can always add a `static_assert` with `std::is_arithmetic`.